assembly/src/docs/20_INSTALL_WINDOWS_WITH_GRAALVM.md

```diff
@@ -0,0 +1,133 @@
+export TORNADO_SDK=$PWD/bin/sdk              ## This directory will be automatically generated during Tornado compilation
+CMAKE_FILE=$(where cmake | head -n 1)
+export CMAKE_ROOT=${CMAKE_FILE%\\*\\*}
+```
+There are only 2 places you should adjust:
+1. `JAVA_HOME` path that points to your Graal installation
+2. `CUDA_PATH` pointing to your NVIDIA GPU Computing Toolkit (CUDA) - this one is necessary only for builds with PTX backend.
+
+## Running build
+Start `<MSYS2>/mingw64.exe` terminal, navigate to the `<TornadoVM>`  directory and execute build:
+```bash
+cd D:/MyProjects/TornadoVM
+source etc/sources.env
+make graal-jdk-8 BACKEND=ptx,opencl
+```
+The latest command, `make`, depends on the Graal version you are using:
+1. For Graal JDK 8 use `make graal-jdk-8 BACKEND=ptx,opencl` (as above)
+2. For Graal JDK 11 use `make graal-jdk-11 BACKEND=ptx,opencl` 
+
+The `BACKEND` parameter has to be a comma-separated list of `ptx` and `opencl` options. You may build `ptx` only when NVIDIA GPU Computing Toolkit (CUDA) is installed.
+
+## Verifying build
+Don't close `<MSYS2>/mingw64.exe` after build. Run the following command to see that TornadoVM is working:
+```bash
+tornado --devices
+```
+You should see a list of OpenCL and/or CUDA devices available on your system.
+
+Now try to run a simple test. For Graal JDK 11:
+```bash 
+tornado -m tornado.examples/uk.ac.manchester.tornado.examples.compute.MatrixMultiplication2D 512
+```
+
+... for Graal JDK 8:
+```bash 
+tornado uk.ac.manchester.tornado.examples.compute.MatrixMultiplication2D 512
+```
+
+If "sanity checks" from above work, you can execute complete test suite for TornadoVM:
+```bash
+tornado-test.py -V
+```
+
+## Known issues
+1. If you already have MSys2 installed and heavily customized you may experience issues with build or tests. We are suggesting to start with fresh MSys2 installation in this case and follow the instructions above. Most notably, make sure that you have no `mingw-w64-x86_64-python` installed - it prevents Python scripts that execute tests from running. Also, make sure that you have updated all GCC / Make / CMake packages mentioned.
+2. If you see no output from `tornado --devices` this may be either of 2 reasons:
+-- OpenCL / CUDA is misconfigured. Download any third-party tool for OpenCL / CUDA capabilities viewing and check that you can see your devices there. Sometimes order of installation of different OpenCL drivers matters - Intel OpenCL SDK may shadow NVIDIA OpenCL and alike.
+-- You build native code of the library using wrong compiler, most probably you ran `<MSYS2>/msys2.exe` terminal instead of `<MSYS2>/mingw64.exe` . Please re-try with correct terminal (and therefore GCC) version.
+3. If you see JVM crashes or `UnsatisfiedLinkError` or some `Error initializing DLL` during `tornado --devices` execution than it's definitely due to wrong GCC (and hence terminal) version used during build.
```

Comment:
How we can install a specific version of GCC to avoid this problem?

Reply:
We will write above that GCC version should be checked to be 10 or higher

Reply:
But TornadoVM can run with any version >-5.5. My question was about how can we download a specific version with the instructions you provide.

Reply:
We will merely reproduce MSys2 docs. For those, who install MSys2 for the first time all necessary versions (the latest and greatest) will be provided automatically. Those ones, who have existing installation are warned and may either update (and get the newest versions) or selectively update with specifying explicit version according to MSys2 docs (and we will point above which ones are reported to be valid).